seerep-hdf5/seerep-hdf5-fb/src/hdf5-fb-general.cpp

```diff
@@ -11,56 +11,53 @@ Hdf5FbGeneral::Hdf5FbGeneral(std::shared_ptr<HighFive::File>& file, std::shared_
-                             static_cast<const seerep::fb::Boolean*>(attribute->value())->data());
-      }
-      else if (attribute->value_type() == seerep::fb::Datatypes_Integer)
-      {
-        writeAttributeToHdf5(*dataSetPtr, attribute->key()->str(),
-                             static_cast<const seerep::fb::Integer*>(attribute->value())->data());
-      }
-      else if (attribute->value_type() == seerep::fb::Datatypes_Double)
-      {
-        writeAttributeToHdf5(*dataSetPtr, attribute->key()->str(),
-                             static_cast<const seerep::fb::Double*>(attribute->value())->data());
-      }
-      else if (attribute->value_type() == seerep::fb::Datatypes_String)
-      {
-        writeAttributeToHdf5(*dataSetPtr, attribute->key()->str(),
-                             static_cast<const seerep::fb::String*>(attribute->value())->data()->str());
-      }
-      else
-      {
-        std::stringstream errorMsg;
-        errorMsg << "type " << attribute->value_type() << " of attribute " << attribute->key()->c_str()
-                 << " not implemented in hdf5-io.";
-        BOOST_LOG_SEV(m_logger, boost::log::trivial::severity_level::error) << errorMsg.str();
-        throw std::invalid_argument(errorMsg.str());
-      }
+      writeAttributeToHdf5(*dataSetPtr, attribute->key()->str(),
+                           static_cast<const seerep::fb::Boolean*>(attribute->value())->data());
+    }
+    else if (attribute->value_type() == seerep::fb::Datatypes_Integer)
+    {
+      writeAttributeToHdf5(*dataSetPtr, attribute->key()->str(),
+                           static_cast<const seerep::fb::Integer*>(attribute->value())->data());
+    }
+    else if (attribute->value_type() == seerep::fb::Datatypes_Double)
+    {
+      writeAttributeToHdf5(*dataSetPtr, attribute->key()->str(),
+                           static_cast<const seerep::fb::Double*>(attribute->value())->data());
+    }
+    else if (attribute->value_type() == seerep::fb::Datatypes_String)
+    {
+      writeAttributeToHdf5(*dataSetPtr, attribute->key()->str(),
+                           static_cast<const seerep::fb::String*>(attribute->value())->data()->str());
+    }
+    else
+    {
+      std::stringstream errorMsg;
+      errorMsg << "type " << attribute->value_type() << " of attribute " << attribute->key()->c_str()
+               << " not implemented in hdf5-io.";
+      BOOST_LOG_SEV(m_logger, boost::log::trivial::severity_level::error) << errorMsg.str();
+      throw std::invalid_argument(errorMsg.str());
     }
   }
 }
 
 void Hdf5FbGeneral::writeBoundingBoxLabeled(
     const std::string& datatypeGroup, const std::string& uuid,
-    const flatbuffers::Vector<flatbuffers::Offset<seerep::fb::BoundingBoxLabeled>>* boundingboxLabeled)
+    const flatbuffers::Vector<flatbuffers::Offset<seerep::fb::BoundingBoxLabeled>>& boundingboxLabeled)
 {
-  if (boundingboxLabeled && !boundingboxLabeled->size() == 0)
+  if (!boundingboxLabeled.size() == 0)
```

Comment:
same for this: are you sure that this doesn't create a segmentation fault if the boundingboxLabeled vector in the FB msg isn't set?

Reply:
Before calling this function, I checked it the field is set, but it still makes sense to add it here for additional safety.

Reply:
It will cause a segmentation fault, but generally all the functions use references and to call the function you would have dereference a `null pointer` which is undefined behavior. Now the question is, how we want to handle that, let it crash, to force the null checks' before the function is called or also catch it in the function.

Reply:
exactly. For that reason the pointer was the argument for the methode previously. With this the pointer could be a `null pointer` and the check was done in the methode. Otherwise we'd have to check if the funtion is not `null` before each call. 

We cannot avoid the `null pointers`, because the sub messages of the fb message are optional and the pointer to the submessage is `null` if the submessage is not used.

In summary: we have to check either before each methode call, which may be forgotten and create a crash of the server or we check it just once inside of the methode. I'd prefer inside of the methode